pkg/operator/ceph/operator.go

```diff
@@ -215,13 +215,25 @@ func (o *Operator) stopSystemDaemons() error {
 	if o.operatorNamespace == "" {
 		return errors.Errorf("Rook operator namespace is not provided. Expose it via downward API in the rook operator manifest file using environment variable %q", k8sutil.PodNamespaceEnvVar)
 	}
+	serverVersion, err := o.context.Clientset.Discovery().ServerVersion()
+	if err != nil {
+		return errors.Wrapf(err, "error getting server version")
+	}
+
+	if serverVersion.Major < csi.KubeMinMajor || serverVersion.Major == csi.KubeMinMajor && serverVersion.Minor < csi.KubeMinMinor {
```

Comment:
We have so many of these complex `if` checks to test for supported version scenarios, and I'm concerned that it's very easy to make a typo in them and mess up the logic. Maybe we should include some reusable version checking code in the non-backend-specific Rook code? Then we could also have good unit tests to make sure we are less at risk of human error in times like this.

```go
type SemVer struct {
	Major  int
	Minor  int
	Bugfix int
}

func (v *SemVer) IsGreaterThan(g SemVer) bool {
	return v.Major > g.Major ||
		(v.Major == g.Major && v.Minor > g.Minor) ||
		(v.Major == g.Major && v.Minor == g.Minor && v.Bugfix > g.Bugfix)
}


// etc for LessThan, LTEQ, GTEQ, EQ
```